src/components/Guilds/Header/WalletButton.tsx

```diff
@@ -0,0 +1,102 @@
+import { observer } from 'mobx-react';
+import { isDesktop } from 'react-device-detect';
+import styled from 'styled-components';
+import WalletModal from 'components/WalletModal';
+import { getChains, injected } from 'provider/connectors';
+import { useEffect, useState } from 'react';
+import { useRpcUrls } from 'provider/providerHooks';
+import { useContext } from '../../../contexts';
+import { Button, IconButton } from '../common/Button';
+import useENSAddress from '../../../hooks/useENSAddress';
+import Avatar from '../Avatar';
+import { shortenAddress } from '../../../utils';
+
+const IconHolder = styled.span`
+  display: flex;
+  justify-content: center;
+  margin-right: 0.3rem;
+
+  img {
+    border-radius: 50%;
+    margin-right: 0;
+  }
+`;
+
+const Web3Status = observer(() => {
+  const {
+    context: { modalStore, providerStore },
+  } = useContext();
+  const [injectedWalletAuthorized, setInjectedWalletAuthorized] =
+    useState(false);
+  const rpcUrls = useRpcUrls();
+
+  useEffect(() => {
+    injected.isAuthorized().then(isAuthorized => {
+      setInjectedWalletAuthorized(isAuthorized);
+    });
+  }, [injected]);
+
+  const { chainId, account } = providerStore.getActiveWeb3React();
+  const { ensName, avatarUri } = useENSAddress(account);
+
+  const toggleWalletModal = () => {
+    modalStore.toggleWalletModal();
+  };
+
+  const switchNetwork = async chain => {

```

Comment:
I think also the network methods of swithNetwork, and get actual network, and whatever is about the network could be in a hook. Because we may need to change networks from other places? may be im wrong too!

Reply:
I'll handle a refactor of how we handle this in my next PR. Thinking of separating our Guilds web3 implementation from the DXvote one, so we can fix a bunch of issues.